subsys/net/lib/openthread/platform/radio.c

```diff
@@ -844,3 +844,23 @@ otError otPlatRadioSetTransmitPower(otInstance *aInstance, int8_t aPower)
 
 	return OT_ERROR_NONE;
 }
+
+#if !CONFIG_OPENTHREAD_EUI64_CUSTOM_SOURCE
+void otPlatRadioGetIeeeEui64(otInstance *aInstance, uint8_t *aIeeeEui64)
+{
+    ARG_UNUSED(aInstance);
+
+    uint64_t factoryAddress;
+    uint32_t index = 0;
+
+    /* Set the MAC Address Block Larger (MA-L) formerly called OUI. */
+    aIeeeEui64[index++] = (CONFIG_OPENTHREAD_STACK_VENDOR_OUI >> 16) & 0xff;
+    aIeeeEui64[index++] = (CONFIG_OPENTHREAD_STACK_VENDOR_OUI >> 8) & 0xff;
+    aIeeeEui64[index++] = CONFIG_OPENTHREAD_STACK_VENDOR_OUI & 0xff;
+
+    /* Use device identifier assigned during the production. */
+    factoryAddress = (uint64_t)NRF_FICR->DEVICEID[0] << 32;
```

Comment:
The radio is not necessarily the nrf device, this will not work for any other radio.

Reply:
Agreed, OpenThread just copies the LL address configured on the network interface, if we want to customize it I think it should happen at this level:
https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/ieee802154/ieee802154_nrf5.c#L65
IMO it should be fine to customize eui64 based on the link-layer selected in there.

Reply:
It was decided to implement such feature only for NRFs, so I moved everything to the nrf Kconfig and ieee802154_nrf5.c file.